roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -102,6 +102,18 @@ impl JDsMempool {
             .ok_or(JdsMempoolError::NoClient)?;
 
         let mempool: Vec<String> = client.get_raw_mempool().await?;
+
```

Comment:
you may want to change this function with something like this
```
    pub async fn update_mempool(self_: Arc<Mutex<Self>>) -> Result<(), JdsMempoolError> {

        let mut new_jds_mempool: HashMap<Txid, Option<Transaction>> = self_.safe_lock(|x| x.mempool.clone())?;
        // the fat transactions in the jds-mempool are those declared by some downstream and we
        // don't want to remove them, but we can get rid of the others
        // TODO remove this clone!!!!
        for (key, val) in new_jds_mempool.clone().iter() {
            if let None = val {
                new_jds_mempool.remove(key);
            }
        }

        let client = self_
            .safe_lock(|x| x.get_client())?
            .ok_or(JdsMempoolError::NoClient)?;
        let node_mempool: Vec<String> = client.get_raw_mempool().await?;
        // here we add all the new transactions
        for id in &node_mempool {
            let key_id = Txid::from_str(id)
                .map_err(|err| JdsMempoolError::Rpc(RpcError::Deserialization(err.to_string())))?;
            // not sure if correct, check it!
            new_jds_mempool.entry(key_id).or_insert(None);
        }

        if new_jds_mempool.is_empty() {
            Err(JdsMempoolError::EmptyMempool)
        } else {
            let _ = self_.safe_lock(|x| x.mempool = new_jds_mempool);
            Ok(())
        }
    }

```
this because
1. keep the fat txs which are those that appear in some DeclareMiningJob
2. get rid of all other thin transactions
3. add the new transactions retrieved from the bitcoin node

Reply:
Hey @NonsoAmadi10, have you implemented this suggestion from @lorbax?

Reply:
In implementing the TODO for removing the clone, is it not the same thing as using the retain method, keeping only values in the new_jds_mempool that are some which removes the need for cloning again as this way:

`  new_jds_mempool.retain(|_, val| val.is_some());`